src/PKSim.Core/Snapshots/Mappers/ExtendedPropertyMapper.cs

```diff
@@ -34,20 +35,60 @@ private Task<ModelExtendedProperty> mapExtendedProperty<T>(SnapshotExtendedPrope
             Name = snapshot.Name,
             Value = propertyValue
          };
-         addOptionsToList(snapshot.ListOfValues, option => extendedProperty.AddToListOfValues(convertToTypeFunc(option)));
+         addOptionsToList(snapshot.ListOfValues, option => extendedProperty.AddToListOfValues(convertToTypeFunc(option.ToString())));
          return Task.FromResult<ModelExtendedProperty>(extendedProperty);
       }
 
+      private ExtendedPropertyType mapExtendedPropertyType(Type type)
+      {
+         if (type == typeof(int))
+            return ExtendedPropertyType.Integer;
+
+         if (type == typeof(double))
+            return ExtendedPropertyType.Double;
+
+         if (type == typeof(bool))
+            return ExtendedPropertyType.Boolean;
+
+         return ExtendedPropertyType.String;
+      }
+
       public override Task<ModelExtendedProperty> MapToModel(SnapshotExtendedProperty snapshot, SnapshotContext snapshotContext)
       {
-         if (double.TryParse(snapshot.Value.ToString(), out var doubleResult))
-            return mapExtendedProperty(snapshot, option => double.Parse(option.ToString()), doubleResult);
+         var snapshotType = snapshot.Type;
+         var valueAsString = snapshot.Value.ToString();
 
-         if (bool.TryParse(snapshot.Value.ToString(), out var boolResult))
-            return mapExtendedProperty(snapshot, option => bool.Parse(option.ToString()), boolResult);
+         if (snapshotType.HasValue)
+            return mapExtendedPropertyBasedOnType(snapshot, snapshotType.Value, valueAsString);
 
+         return mapExtendedPropertyBasedOnValue(snapshot, valueAsString);
+      }
 
-         return mapExtendedProperty(snapshot, option => option.ToString(), snapshot.Value.ToString());
+      private Task<ModelExtendedProperty> mapExtendedPropertyBasedOnType(ExtendedProperty snapshot, ExtendedPropertyType snapshotType, string valueAsString)
+      {
+         switch (snapshotType)
```

Comment:
@Yuri05 New way.
This should fail if someone messes around with the snamsphot as we are parsing without trying